design-proposals/golden-image-delivery-and-update-pipeline.md

```diff
@@ -0,0 +1,275 @@
+
+## Common templates: DataSources as DataVolume Source
+
+The common templates shipped with SSP need to have their DataVolumeTemplate sections transition to using a DataSource rather than the inlined PVC source field. This allows the DataImportCron to seamlessly update the underlying PVC the corresponding DataSource points to, which then ensures that the common templates are always using the most up-to-date disk images.
+
+## HCO: Integration with SSP Operator
+
+In order to leverage golden disk image auto delivery/updates at the HCO level, the HCO is responsible for configuring DataImportCrons on the SSP operator’s CR.
+
+For instance, if the HCO wants to develop a feature for automatically sourcing in fedora golden images into a PVC to be used by the SSP's common templates, then the HCO needs to configure the location where the fedora image is sourced from on the SSP CR. 
+
+This document doesn't include within its scope how the HCO should expose enabling the image auto delivery/update mechanism on the HCO API or where the containerDisk images are sourced.
+
+## OCP UI: Integration with DataSources
+
+The UI currently detects if a template has a boot source by checking if the DataVolumeTemplate’s source PVC is present. The UI needs to now be capable of checking if a DataVolumeTemplate’s DataSource has the a ready condition equal to 'true'.
+
+## Scalability
+
+- DataImportCrons have the potential to exhaust storage resources. This can be mitigated through a garbage collection policy for out-of-date disk volumes. 
+- A large amount of DataImportCrons triggering at the exact same interval could impact network resources during the import. This can be mitigated by introducing a jitter factor into the recurring poll interval.
+
+## Update/Rollback Compatibility
+
+### Transitioning the CommonTemplates to DataSources
+
+SSP operator must make image pipelines an opt-in feature to ensure existing boot images are not overwritten.
+
+To handle the default (non auto update/delivery) use case during update, the SSP operator will create corresponding DataSources for every DataSource referenced in the DataVolumeTemplate sections present in the deployed common templates. These default DataSources will point to a PVC name that matches the PVC name used before DataSources were introduced into the common templates.
+
+Here's an example of how this works.
+
+**Before Update:** If the common templates had a DataVolumeTemplate that looked like the following before DataSources were introduced.
+```yaml
+dataVolumeTemplates:
+- apiVersion: cdi.kubevirt.io/v1beta1
+  kind: DataVolume
+  metadata:
+    name: fedora
+  spec:
+    pvc:
+      accessModes:
+        - ReadWriteMany
+      resources:
+        requests:
+          storage: 30Gi
+    source:
+      pvc:
+        name: fedora
+        namespace: golden-images
+```
+
+
+**After Update:** The common template would have a DataVolumeTemplate that looks like the following after the update
```

Comment:
What about existing custom templates created from common templates? Should this change apply to these as well?